src/test/java/org/cactoos/scalar/AndInThreadsTest.java

```diff
@@ -163,4 +169,170 @@ public void testFunc() throws Exception {
+            1, 2
+        ).value();
+        MatcherAssert.assertThat(
+            list,
+            Matchers.containsInAnyOrder(1, 2)
+        );
+    }
+
+    @Test
+    public void testExecServiceProcIterator() throws Exception {
+        final List<Integer> list = Collections.synchronizedList(
+            new ArrayList<Integer>(2)
+        );
+        new AndInThreads(
+            Executors.newSingleThreadExecutor(),
+            new Proc.NoNulls<Integer>(list::add),
+            Arrays.asList(1, 2).iterator()
+        ).value();
+        MatcherAssert.assertThat(
+            list,
+            Matchers.containsInAnyOrder(1, 2)
+        );
+    }
+
+    @Test
+    public void testExecServiceProcIterable() throws Exception {
+        final List<Integer> list = Collections.synchronizedList(
+            new ArrayList<Integer>(2)
+        );
+        final ExecutorService service = Executors.newSingleThreadExecutor();
+        new AndInThreads(
+            service,
+            new Proc.NoNulls<Integer>(list::add),
+            Arrays.asList(1, 2)
+        ).value();
+        MatcherAssert.assertThat(
+            list,
+            Matchers.containsInAnyOrder(1, 2)
+        );
+    }
+
+    @Test
+    public void testExecServiceScalarBooleans() throws Exception {
+        MatcherAssert.assertThat(
+            new AndInThreads(
+                Executors.newSingleThreadExecutor(),
+                new Scalar.NoNulls<Boolean>(
+                    new Constant<Boolean>(true)
+                ),
+                new Scalar.NoNulls<Boolean>(
+                    new Constant<Boolean>(false)
+                )
+            ).value(),
+            Matchers.equalTo(false)
+        );
+    }
+
+    @Test
+    public void testExecServiceIterableScalarBoolean() throws Exception {
+        MatcherAssert.assertThat(
+            new AndInThreads(
+                Executors.newSingleThreadExecutor(),
+                new ListOf<Scalar<Boolean>>(
+                    new Scalar.NoNulls<Boolean>(
+                        new Constant<Boolean>(true)
+                    ),
+                    new Scalar.NoNulls<Boolean>(
+                        new Constant<Boolean>(false)
+                    )
+                )
+            ).value(),
+            Matchers.equalTo(false)
+        );
+    }
+
+    @Test
+    public void testExecServiceIteratorScalarBoolean() throws Exception {
+        MatcherAssert.assertThat(
+            new AndInThreads(
+                Executors.newSingleThreadExecutor(),
+                new ListOf<Scalar<Boolean>>(
+                    new Scalar.NoNulls<Boolean>(
+                        new Constant<Boolean>(true)
+                    ),
+                    new Scalar.NoNulls<Boolean>(
+                        new Constant<Boolean>(false)
+                    )
+                ).iterator()
+            ).value(),
+            Matchers.equalTo(false)
+        );
+    }
+
 }
```

Comment:
@driver733 I see multiple tests for logical conjunction between booleans, but only for ``true``/``false``. What about ``true``/``true``, ``false``/``false`` and ``false``/``true``?